PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,339 @@
+  template <typename T1>
+  void GetQvecDtracks(const T1& cand,
+                      std::vector<float>& tracksQx,
+                      std::vector<float>& tracksQy,
+                      float& ampl)
+  {
+    // TODO: add possibility to consider different weights for the tracks, at the only pT is considered;
+    float pXtrack0 = cand.pxProng0();
+    float pYtrack0 = cand.pyProng0();
+    float pTtrack0 = cand.ptProng0();
+    float phiTrack0 = TMath::ATan2(pYtrack0, pXtrack0);
+    float pXtrack1 = cand.pxProng1();
+    float pYtrack1 = cand.pyProng1();
+    float pTtrack1 = cand.ptProng1();
+    float phiTrack1 = TMath::ATan2(pYtrack1, pXtrack1);
+    float pXtrack2 = cand.pxProng2();
+    float pYtrack2 = cand.pyProng2();
+    float pTtrack2 = cand.ptProng2();
+    float phiTrack2 = TMath::ATan2(pYtrack2, pXtrack2);
+
+    tracksQx.push_back(TMath::Cos(harmonic * phiTrack0) * pTtrack0 / ampl);
+    tracksQy.push_back(TMath::Sin(harmonic * phiTrack0) * pTtrack0 / ampl);
+    tracksQx.push_back(TMath::Cos(harmonic * phiTrack1) * pTtrack1 / ampl);
+    tracksQy.push_back(TMath::Sin(harmonic * phiTrack1) * pTtrack1 / ampl);
+    tracksQx.push_back(TMath::Cos(harmonic * phiTrack2) * pTtrack2 / ampl);
+    tracksQy.push_back(TMath::Sin(harmonic * phiTrack2) * pTtrack2 / ampl);
+  }
+
+  /// Compute the delta psi in the range [0, pi/harmonic]
+  /// \param psi1 is the first angle
+  /// \param psi2 is the second angle
+  /// \note Ported from AliAnalysisTaskSECharmHadronvn::GetDeltaPsiSubInRange
+  float GetDeltaPsiInRange(float psi1, float psi2)
+  {
+    float deltaPsi = psi1 - psi2;
+    if (deltaPsi > TMath::Pi() / harmonic) {
+      if (deltaPsi > 0.)
+        deltaPsi -= 2. * TMath::Pi() / harmonic;
+      else
+        deltaPsi += 2. * TMath::Pi() / harmonic;
+    }
+    return deltaPsi;
+  }
+
+  /// Fill THnSparse
+  /// \param mass is the invariant mass of the candidate
+  /// \param pt is the transverse momentum of the candidate
+  /// \param cent is the centrality of the collision
+  /// \param cosNPhi is the cosine of the n*phi angle
+  /// \param cosDeltaPhi is the cosine of the n*(phi - evtPl) angle
+  /// \param sp is the scalar product
+  /// \param evtPlReso is the event plane resolution
+  /// \param outputMl is the ML score
+  void fillThn(float mass,
+               float pt,
+               float cent,
+               float cosNPhi,
+               float cosDeltaPhi,
+               float sp,
+               float outputMl)
+  {
+    registry.fill(HIST("hSparseFlowCharm"), mass, pt, cent, cosNPhi, cosDeltaPhi, sp, outputMl);
+  }
+
+  /// Get the centrality
+  /// \param collision is the collision with the centrality information
+  /// \param centDetector is the detector used for the centrality estimation
+  float GetCent(CollsWithQvecs::iterator const& collision, int centDetector)
+  {
+    float cent = -999.;
+    switch (centDetector) {
+      case 0: // V0M
+        cent = collision.centFV0A();
+        break;
```

Comment:
It would be clearer to define `enum` for these values. If there are changes, it will be easier to update the `centDetector` documentation by checking the `enum` definition instead of looking inside the code which uses it.

Reply:
Ok, I will do something like it is done at skimming level then [https://github.com/AliceO2Group/O2Physics/blob/2fdafcea42ac73d894464919dc4b4643fa3e0070/PWGHF/TableProducer/trackIndexSkimCreator.cxx#L83]